elephant/spade.py

```diff
@@ -977,19 +971,19 @@ def _fast_fca(context, min_c=2, min_z=2, max_z=None,
                 np.array(intent) % winlen)] += 1
     if report == 'a':
         return concepts
-    else:
-        del concepts
-        # returning spectrum
-        if report == '#':
-            for (z, c) in np.transpose(np.where(spec_matrix != 0)):
-                spectrum.append((z + 1, c + 1, int(spec_matrix[z, c])))
 
-        if report == '3d#':
-            for (z, c, l) in np.transpose(np.where(spec_matrix != 0)):
-                spectrum.append(
-                    (z + 1, c + 1, l, int(spec_matrix[z, c, l])))
-        del spec_matrix
-        return spectrum
+    del concepts
```

Comment:
[not now]
We need to discuss all these `del x` statements that, in general, should be avoided.
I bet all of them came from C++, where it makes sense. But not in python.